app/main.py

```diff
@@ -2,5 +1,17 @@
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    storage = {}
+
+    def wrapper(*args: Any) -> Any:
+        storage_keys = storage.keys()
+        if args not in storage_keys:
```

Comment:
Consider handling cases where `args` might not be hashable. Since `args` is used as a dictionary key, it must be hashable. You might want to add a try-except block to handle potential `TypeError` exceptions.